webserver/server/app/models/transaction/sqlHistoryForAddresses.sql

```diff
@@ -47,9 +47,15 @@ SELECT "Transaction".id,
         "Block".epoch,
         "Block".slot,
         "Block".era,
-        "Block".height
+        "Block".height,
+        "TransactionMetadata".payload AS metadata,
+        json_agg(DISTINCT "Address".PAYLOAD) input_addresses
```

Comment:
I'm slightly concerned this may cause a large performance regression on the tx history endpoint for use-cases that don't care about the metadata

Reply:
Well... But I'm not sure adding a dedicated endpoint for it it's a good idea neither. Potentially we can make this opt-in, although I'm not quite sure yet on what's the best way of doing it at the query level. But I could try to setup a benchmark of some kind to figure out the impact.